tt_eager/tt_dnn/op_library/reduce/reduce_op.cpp

```diff
@@ -262,16 +264,47 @@ Tensor sum(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_
             formatted_input_tensor = AutoFormat::format_input_tensor(input_tensor, device, input_tensor_pad_shape, 0.0, Layout::TILE);
         }
         Tensor output = transpose(input_tensor, 0, -2, output_mem_config);
-        output = sum(output, 2, output_mem_config);
+        output = reduce_on_dim<OpKind>(output, 2, output_mem_config);
         output = transpose(output, 0, -2, output_mem_config);
         return AutoFormat::format_output_tensor(output, out_shape, device, Layout::TILE);
     }
 }
 
-Tensor global_sum(Tensor& val, const MemoryConfig& output_mem_config) {
+
+Tensor sum(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_config) {
+    TT_FATAL( dim >= 0 && dim <= 3, "dimension have to be 0-3 only corresponding to N,C,H,W");
```

Comment:
Does this assert need to be in each of these functions if it's already in `reduce_on_dim`?